contribs/gnodev/pkg/dev/node_state.go

```diff
@@ -0,0 +1,140 @@
+package dev
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gnolang/gno/contribs/gnodev/pkg/events"
+	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+// Save the current state as initialState
+func (n *Node) SaveCurrentState(ctx context.Context) error {
+	n.muNode.RLock()
+	defer n.muNode.RUnlock()
+
+	// Get current blockstore state
+	state, err := n.getState(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to save state: %s", err.Error())
+	}
+
+	n.initialState = state[:n.currentStateIndex]
+	return nil
+}
+
+// Export the current state as list of txs
+func (n *Node) ExportCurrentState(ctx context.Context) ([]std.Tx, error) {
+	n.muNode.RLock()
+	defer n.muNode.RUnlock()
+
+	// Get current blockstore state
```

Comment:
This fetches the state, not the `n.getBlockstoreState`

Reply:
It's actually the result of `n.getBlockstoreState` that's why I put a comment here. `getState` only caches the result on multiple calls to avoid overhead